common/js/src/popup-window/popup-opener.js

```diff
@@ -92,8 +92,9 @@ GSC.PopupOpener.createWindow = function(url, windowOptions, opt_data) {
   /** @preserveTry */
   try {
     if (GSC.Packaging.MODE === GSC.Packaging.Mode.APP) {
+      const loaclUrl = url.replace("chrome-extension://khpfeaanjngmcnplbdlpegiifgpfgdco", "");
```

Comment:
We shouldn't hardcode the "chrome-extension://" or the "khpfeaanjngmcnplbdlpegiifgpfgdco" parts here - it's intended as a generic code that should work regardless of these concrete details. Also low-level string replacement will break in case the searched pattern occurs in other parts of the URL - say, in the URL parameters.

I think the original root cause issue is that our API in this file became asymmetric in #440: `runModalDialog()` takes a "hostless" URL (I don't know what's the proper term for this), but then passes a URL with host to `createWindow()`. So can we fix `runModalDialog()` to create a "hostless" URL and pass it further? I don't know if JavaScript has a nicer helper for this, but it can be something like

```
modifiedUrl.pathname + modifiedUrl.search + modifiedUrl.hash
```